WordPress/gutenberg
# Heading block: Add support for writingMode

## What?

Adds block support for typography writingMode to the heading block.

## Why?

To enable vertical text for headings.
This is needed for [one of the block patterns in Twenty Twenty-Four.](https://github.com/WordPress/twentytwentyfour/issues/122)

## Testing Instructions
Activate a block theme.
The writing mode setting is disabled by default; set it to true in theme.json settings > typography > writingMode:

```
	"settings": {
		"typography": {
			"writingMode": true
		}
	}
```

Create a new post or page, and add a heading block.
Enable the vertical text orientation option in the typography panel.
Save and view the front.
Confirm that the text is now vertical in the editor and the front.

Use the block toolbar and set the text alignment to right (ltr) or left (rtl).
Confirm that the text is now "upside down": the first letter in the heading is at the bottom.

## Screenshot
<img width="1508" alt="a heading block placed in the block editor, with writingmode enabled" src="https://github.com/WordPress/gutenberg/assets/7422055/ce7ff0c4-25f1-480d-988c-d868b89bc6c1">
